appshell/appshell_extensions_platform.cpp

```diff
@@ -0,0 +1,89 @@
+#include "appshell/appshell_extensions_platform.h"
+#include <unicode/ucsdet.h>
+#include <unicode/ucnv.h>
+#include <memory>
+
+CharSetDetect::CharSetDetect() {
+	charsetDetector_ = NULL;
+	icuError = U_ZERO_ERROR;
+	charsetDetector_ = ucsdet_open(&icuError);
+	if (U_FAILURE(icuError))
+		throw "Failed to open detector";
+}
+
+CharSetDetect::~CharSetDetect() {
+	if (charsetDetector_) {
+		ucsdet_close(charsetDetector_);
+	}
+}
+
+void CharSetDetect::operator()(const char* bufferData, size_t bufferLength, std::string &detectedCharSet) {
+	detectedCharSet = "";
+	const UCharsetMatch* charsetMatch_;
+
+	// send text
+	ucsdet_setText(charsetDetector_, bufferData, bufferLength, &icuError);
+	if (U_FAILURE(icuError))
+		throw "Failed to set text";
+
+	// detect language
+	charsetMatch_ = ucsdet_detect(charsetDetector_, &icuError);
+	if (U_FAILURE(icuError))
+		throw "Failed to detect CharSet";
+
+	const char* detectedCharsetName = ucsdet_getName(charsetMatch_, &icuError);
+	detectedCharSet = detectedCharsetName;
+
+	// Get Language Name
+	const char* detectedLanguage = ucsdet_getLanguage(charsetMatch_, &icuError);
```

Comment:
I do not understand where are `detectedLanguage` and `detectionConfidence` getting used. you may have to clean these up if you had added for debugging purpose alone.